aptos-move/aptos-validator-interface/src/lib.rs

```diff
@@ -95,37 +106,71 @@ pub trait AptosValidatorInterface: Sync {
             let addr = *validator_info.account_address();
             result.push((
                 addr,
-                self.get_account_state_by_version(addr, version)?
+                self.get_account_state_by_version(addr, version)
+                    .await?
                     .ok_or_else(|| anyhow!("validator {:?} doesn't exist", addr))?,
             ));
         }
         Ok(result)
     }
 }
 
-pub struct DebuggerStateView<'a> {
-    db: &'a dyn AptosValidatorInterface,
+pub struct DebuggerStateView {
+    query_sender: Mutex<UnboundedSender<(StateKey, Version)>>,
+    query_receiver: Mutex<Receiver<Result<Option<StateValue>>>>,
     version: Option<Version>,
 }
 
-impl<'a> DebuggerStateView<'a> {
-    pub fn new(db: &'a dyn AptosValidatorInterface, version: Option<Version>) -> Self {
-        Self { db, version }
+async fn handler_thread<'a>(
+    db: Arc<dyn AptosValidatorInterface + Send>,
+    mut thread_receiver: UnboundedReceiver<(StateKey, Version)>,
+    thread_sender: Sender<Result<Option<StateValue>>>,
+) {
+    loop {
+        let (key, version) = if let Some((key, version)) = thread_receiver.recv().await {
+            (key, version)
+        } else {
+            break;
+        };
+        let val = db.get_state_value_by_version(&key, version - 1).await;
+        thread_sender.send(val).unwrap();
+    }
+}
+
+impl DebuggerStateView {
+    pub fn new(db: Arc<dyn AptosValidatorInterface + Send>, version: Option<Version>) -> Self {
+        let (query_sender, thread_receiver) = unbounded_channel();
+        let (thread_sender, query_receiver) = channel();
+
+        tokio::spawn(async move { handler_thread(db, thread_receiver, thread_sender).await });
```

Comment:
this only works if it's inside a tokio context otherwise it'd just panic.
I don't think we need channels for this, just use block_on should be sufficient

Reply:
To problem is that `block_on` would panic if used inside of a tokio context so would be problematic if we want to use this tool in the `aptos-cli`. Maybe I'll provide both versions to support either use cases?

Reply:
hmm, probably we can just check if it's in async context and decide what to do